xonsh/tools.py

```diff
@@ -521,19 +513,16 @@ def always_false(x):
 
 def ensure_string(x):
     """Returns a string if x is not a string, and x if it already is."""
-    if isinstance(x, string_types):
-        return x
-    else:
-        return str(x)
+    return str(x)
```

Comment:
Please return to the original.  No need to copy the string if it is already a string.  You should get the same object back.